fuse/eval/metrics/metrics_common.py

```diff
@@ -83,23 +83,30 @@ def __init__(
         pre_collect_process_func: Optional[Callable] = None,
         post_collect_process_func: Optional[Callable] = None,
         batch_pre_collect_process_func: Optional[Callable] = None,
+        batch_post_collect_process_func: Optional[Callable] = None,
+        post_keys_to_collect: Optional[List[str]] = None,
         **keys_to_collect: Dict[str, str],
     ):
         """
         :param pre_collect_process_func: Optional callable - the callable will get as an input a sample_dict and can preprocess it if required before collection.
                                          Consider using batch_pre_collect_process_func instead to optimize the running time when working with large batch size.
         :param post_collect_process_func: Optional callable - custom process func - used to evaluate in a sample level and keep only the result.
                                           Typically used in methods such as segmentation to avoid from storing all the images until the end of the epoch.
-                                          Can return either a single value or dictionary. The returned values will be collected under the name "post_args"
-        :param batch_pre_collect_process_func: Optional callable - the callable will get as an input a batch_dict and can preprocess it if required before collection.
+                                          Returns a dictionary to be aggregated.
+        :param batch_pre_collect_process_func: Optional callable - same as pre_collect_process_func but in a batch level (more efficient)
+        :param batch_post_collect_process_func: Optional callable - same as post_collect_process_func but in a batch level (more efficient)
+        :param post_keys_to_collect: specify the keys you want to collect after post_collect_process. Required only if post_collect_process_func or batch_post_collect_process_func are specified.
+                                     if None, will aggregate list of post_collect_process_func returned values
         :param keys_to_collect: specify the keys you want to collect from the source data
         """
         super().__init__()
         # store input
         self._pre_collect_process_func = pre_collect_process_func
         self._post_collect_process_func = post_collect_process_func
         self._batch_pre_collect_process_func = batch_pre_collect_process_func
+        self._batch_post_collect_process_func = batch_post_collect_process_func
```

Comment:
Maybe we should throw an error if a user supplies some batch arguments as well as samples one? for instance `pre_collect_process_func` and `batch_post_collect_process_func`